src/Documentation/grains/transactions.md

```diff
@@ -3,82 +3,110 @@ layout: page
-
-The TransactionsConfiguration contains the following configurable values:
-* TransactionIdAllocationBatchSize - To avoid a storage call on every transaction start, transaction Ids are allocated in batches. This is the number of new transaction Ids allocated per transaction Id generation request (storage call).
-* AvailableTransactionIdThreshold - A new batch of transaction Ids will be automatically allocated if the available ids drop below this threshold.
-* TransactionRecordPreservationDuration - How long to preserve a transaction record in the TM memory after the transaction has completed.  This is used to answer queries about the outcome of the transaction.
+Orleans transactions are opt-in.
+A silo must be configured to use transactions.
+If it is not, any calls to transactional methods on grains will receive an `OrleansTransactionsDisabledException`.
+To enable transactions on a silo, call `UseTransactions()` on the silo host builder.
+`var builder = new SiloHostBuilder().UseTransactions();`
 
-It is suggested that you use the defaults for these values unless they demonstrate themselves as insufficient.
+### Transactional State Storage
 
-### Transaction Log
-
-The transaction manager must persist transaction results for recoverability purposes.  To support a range of backend storage solutions an abstraction has been provided for log storage.  We currently support an azure storage version of the log storage, and an in-memory version for development and test purposes.  This can be programmatically configured on the silo host builder using UseAzureTransactionLog or UseInMemoryTransactionLog.
-
-### Transactional State
-
-To support various transactional storage patterns the grain facing interface and supporting logic is pluggable.  We currently only support ITransactionalState.  This can be programmatically enabled on the silo host builder using UseTransactionalState.  To use a transactional state, a storage provider needs also be configured.
+To use transactions, the user needs to configure a data store.
+To support various data stores with transactions, the storage abstraction `ITransactionalStateStorage` has been introduced.
+This abstraction is specific to the needs of transactions, unlike generic grain storage (`IGrainStorage`).
+To use transaction-specific storage, the user can configure their silo using any implementation of `ITransactionalStateStorage`, such as Azure (`AddAzureTableTransactionalStateStorage`).
 
 Example:
 
 ``` csharp
 var builder = new SiloHostBuilder()
-    [...]
-    .AddAzureTableGrainStorageAsDefault(options => options.ConnectionString = ”YOUR_STORAGE_CONNECTION_STRING”)
-    .UseInClusterTransactionManager()
-    .UseAzureTransactionLog(options => options.ConnectionString = ”YOUR_STORAGE_CONNECTION_STRING”) 
-    .UseTransactionalState();
-
-var host = builder.Build();
-await host.StartAsync();
+    .AddAzureTableTransactionalStateStorage(“TransactionStore”, options =>
```

Comment:
The quotes around TransactionStore here are smart quotes. We need to change them to regular quotes, `"`

Reply:
Done.

>    .AddAzureTableTransactionalStateStorage("TransactionStore", options =>